core/src/main/java/com/amplifyframework/core/DeveloperMenuActivity.java

```diff
@@ -15,16 +15,22 @@
 
 package com.amplifyframework.core;
 
-import android.app.Activity;
 import android.os.Bundle;
-import android.view.KeyEvent;
 import android.view.View;
+import androidx.fragment.app.FragmentActivity;
+import androidx.navigation.NavController;
+import androidx.navigation.Navigation;
+import androidx.navigation.ui.AppBarConfiguration;
+import androidx.navigation.ui.NavigationUI;
 
 /**
  * This is the activity to display the developer menu.
  */
-public final class DeveloperMenuActivity extends Activity {
-    // The parent layout for the developer menu.
+public final class DeveloperMenuActivity extends FragmentActivity {
+
+    // Indicates whether this activity has started.
+    private static boolean activityStarted;
+    // Parent layout for the developer menu.
     private View devMenuLayout;
     // Detect and handle shake events.
     private ShakeDetector detector;
```

Comment:
Hm, you now have two instances of the `ShakeDetector`, one in `Amplify`, and one in this Activity. Ideally, you could have one detector that broadcast to multiple listeners. Instead of the simple callback mechanism we are using right now, you could use `Amplify.Hub` instead. Then, just listen for a hub event in both locations -- from one detector instance.

Reply:
After rewriting the code, I now have only one instance of ShakeDetector (in DeveloperMenuManager). I also tried using Amplify.Hub, but that would require adding a new CategoryType for the developer menu in the HubChannel enum.